include/s3select_functions.h

```diff
@@ -434,8 +434,17 @@ struct _fn_to_int : public base_function
 
     if (func_arg.type == value::value_En_t::STRING)
     {
+      errno = 0;
       i = strtol(func_arg.str(), &perr, 10) ;  //TODO check error before constructor
+      if ((errno == ERANGE && (i == LONG_MAX || i == LONG_MIN)) || (errno != 0 && i == 0)) {
+        throw base_s3select_exception("converted value would fall out of the range of the result type!");
+        return false;   
     }
+    if (perr == func_arg.str()) {
+      throw base_s3select_exception("No digits found!");
```

Comment:
not correct.
upon /perr/ is not null (point on some string) it means it has some error while parsing.
the following statement will pass  through with 12 as result,
**cast("12hhd" as int)** 

it means that random string may process as number.